eng/common/scripts/logging.ps1

```diff
@@ -1,40 +1,93 @@
-function Test-SupportsDevOpsLogging()
-{
-    return ($null -ne $env:SYSTEM_TEAMPROJECTID)
+function Test-SupportsDevOpsLogging() {
+  return ($null -ne $env:SYSTEM_TEAMPROJECTID)
 }
 
-function LogWarning
-{
-    if (Test-SupportsDevOpsLogging)
-    {
-        Write-Host "##vso[task.LogIssue type=warning;]$args"
-    }
-    else
-    {
-        Write-Warning "$args"
-    }
+function Test-SupportsGitHubLogging() {
+  return ($null -ne $env:GITHUB_ACTIONS)
 }
 
-function LogError
-{
-    if (Test-SupportsDevOpsLogging)
-    {
-        Write-Host "##vso[task.LogIssue type=error;]$args"
-    }
-    else
-    {
-        Write-Error "$args"
-    }
+function LogInfo {
+  Write-Host "$args"
+}
+
+function LogWarning {
+  if (Test-SupportsDevOpsLogging) {
+    Write-Host ("##vso[task.LogIssue type=warning;]$args" -replace "`n", "%0D%0A")
+  }
+  elseif (Test-SupportsGitHubLogging) {
+    Write-Warning ("::warning::$args" -replace "`n", "%0D%0A")
+  }
+  else {
+    Write-Warning "$args"
+  }
 }
 
-function LogDebug
+function LogSuccess {
+  $esc = [char]27
```

Comment:
What happens if these show up in DevOps or github logs? I suspect these might only interpreted locally in a console window.

Reply:
I have verified it works on DevOps and GitHub Actions

Reply:
What does it look like there? Does it display those chars?

Reply:
https://github.com/chidozieononiwu/azure-sdk-tools/actions/runs/11603573286/job/32310793801#step:3:14
It just colors the log green.

Reply:
Ok does that work in DevOps logs as well?

Reply:
Yes it does